### DISCUSSION TITLE
How to Update the Container and Zigbee2Mqtt in LXC Container?

### DISCUSSION THREAD

**Discussion by manoloxxl** [tteck/Proxmox#2083] [Help Needed] [answered]
Hi,

I am wondering how i should update the Container and also the Zigbee2Mqtt GUI best way?

As i usual do in Alpine Edge i just run "apk upgrade --available && sync" . During this zigbee2mqtt was updated. 

I also followed the instructions from [zigbee2mqtt doc](https://www.zigbee2mqtt.io/guide/installation/01_linux.html#for-later-update-zigbee2mqtt-to-the-latest-version). But the command "cd /opt/zigbee2mqtt" just show a "-ash: cd: can't cd to /opt/zigbee2mqtt: No such file or directory"

Also i followed [the discussion here](https://github.com/tteck/Proxmox/discussions/1900) as there seems to be a new update function in the scripts (i was not able to find any docs for this function). I followed the instructions to create the update function but "update" just shows "wget: unrecognized option: L"

So, maybe can anybody tell me, for the rest of my machines, what will be the best way to update the images? Just run the apk upgrade --available && sync or is there a "better" way?

Best Regards
Manuel

**Comment by tteck** [comment, marked as answer]
To update the Alpine-Zigbee2MQTT application, `apk update && apk upgrade` is the only way. 
https://pkgs.alpinelinux.org/packages?name=zigbee2mqtt&branch=edge&repo=&arch=x86_64&maintainer=

**Reply by PatrikStaehli** [reply]
It seems like the Alpine zigbee2mqtt package is not upgraded regularly (to date still on 1.34, current release would be 1.35.2).
Is the debian version also relying on 3rd party packaged zigbee2mqtt, or is that using the original source (therefore allowing to upgrade as soon as a new version is available)?
If this is a known drawback of the alpine version, a remark on the scripts page would be great. I prefer small containers, but not at the cost of not being able to upgrade to the latest version...
Thanks for the great work!

**Reply by tteck** [reply]
Yes, the Debian version updates from source. No waiting.

**Reply by PatrikStaehli** [reply]
Great, thanks for the info.
I'll switch to debian then, even though I'll probably have to jump through hoops again as I prefer to run unprivileged containers.
